core/src/main/scala/dogs/BiPartition.scala

```diff
@@ -0,0 +1,36 @@
+/**
+  * Created by anicolaspp on 4/10/16.
+  */
+
+package dogs
+
+import Predef._
+import simulacrum.typeclass
+import scala.{inline,Iterable}
+import java.lang.{String,StringBuilder}
+import scala.annotation.{tailrec}
+import dogs.syntax.birds._
+import cats._
+
+
+@typeclass trait BiPartition[A] {
+  def partition(predicate: A => Boolean): (List[A], List[A])
```

Comment:
It is true, it should be able to get an structure F[A] and transform it into a Partition[A] where BiPartition[A] is an instance of Partition[A].

Having a way to partition structures is important, for instance, in Diet, when you say to diet.intervals that is a partition of Diet. 

If you have a Set you could apply a function to it and get some kind of partition out of it (equivalence classes). 

You could partition a tree in sub-trees, or a graph, in subgraphs. 

I think it would be nice to have, I just need to change its definition.